compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpCompilerPluginUtil.java

```diff
@@ -149,7 +149,7 @@ public static void validateReturnType(SyntaxNodeAnalysisContext ctx, Node node,
         } else if (kind == TypeDescKind.ARRAY) {
             TypeSymbol memberTypeDescriptor = ((ArrayTypeSymbol) returnTypeSymbol).memberTypeDescriptor();
             validateArrayElementTypeInReturnType(
-                    ctx, node, returnTypeStringValue, memberTypeDescriptor, diagnosticCode);
+                    ctx, node, returnTypeStringValue, memberTypeDescriptor, diagnosticCode, isInterceptorType);
```

Comment:
Passing `isInterceptorType` along with the `validateArrayElementTypeInReturnType`, will allow developer to pass an array of `http:NextService` or an array of type which has a member `http:NextService`. So, lets remove that.

Seems like we are passing it in other places as well, like when the type is a map (Line #165). Can you change these as well?

Reply:
Updated

Reply:
Passing https://github.com/dilanSachi/module-ballerina-http/actions/runs/4013386586